### PR TITLE
Fix community account switch behavior

### LINKED ISSUE #1442: Infinite "reload feed" popup in community
The reload feed popup doesn't go away in a community view

### PR DESCRIPTION
**Pull request by EricBAndrews**
>[!NOTE]
>Depends on [this PR](https://github.com/mlemgroup/MlemMiddleware/pull/73)

# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #1442

# Pull Request Information

Fixes an issue where the outdated feed popup didn't work in communities.

- Changing `feedLoader` API is now handled in `outdatedFeedPopup`
- Communities will not show the outdated feed popup if they are in guest mode

### CODE REVIEW

File: Mlem.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2023,7 +2023,7 @@
 				CDE4AC402CA3706400981010 /* XCRemoteSwiftPackageReference "SDWebImageSwiftUI" */,
 				CDE4AC412CA3706F00981010 /* XCRemoteSwiftPackageReference "SDWebImageWebPCoder" */,
 				CD64A9182CA37D63007CA7E6 /* XCRemoteSwiftPackageReference "Gifu" */,
-				CD555F802CE56CB200E763E7 /* XCRemoteSwiftPackageReference "MlemMiddleware" */,
+				CDE1816C2CEAA4E6001D7492 /* XCLocalSwiftPackageReference "../MlemMiddleware" */,
```
Comment (Sjmarf) [inline Mlem.xcodeproj/project.pbxproj]: Reminder

**Review by Sjmarf — CHANGES_REQUESTED** [review]
Code looks good! Found a couple of bugs:

1) When using a lemmy.world guest, tapping "reload" in a `CommunityView` causes the popup to immediately appear again.
2) Previously the outdated feed popup would slide in and out - now it just appears, with no animation. The animation still works fine in `PersonView`, but not in `CommunityView` or the feed.

**Comment by EricBAndrews** [comment]
> Previously the outdated feed popup would slide in and out - now it just appears, with no animation. The animation still works fine in PersonView, but not in CommunityView or the feed.

This was kinda tricky! The cause was the added check in `CommunityView` to avoid presenting the popup on unresolvable communities--this was causing the _outer_ `if canShowPopup` to toggle the entire popup view, bypassing the animation. `canShowPopup` is now simply factored into the `showRefreshPopup` logic.

**Review by Sjmarf — APPROVED** [review]
LGTM 💯

**Pull request merged**